packages/@ourworldindata/utils/src/image.ts

```diff
@@ -62,3 +62,34 @@ export function getFilenameWithoutExtension(
 export function getFilenameAsPng(filename: ImageMetadata["filename"]): string {
     return `${getFilenameWithoutExtension(filename)}.png`
 }
+
+export type SourceProps = {
+    media: string | undefined
+    srcSet: string
+}
+
+/**
+ * When we have a small and large image, we want to generate two <source> elements.
+ * The first will only be active at small screen sizes, due to its `media` property.
+ * The second will be active at all screen sizes.
+ * These props work in conjuction with a `sizes` attribute on the <source> element.
+ */
+export function generateSourceProps(
+    smallImage: ImageMetadata | undefined,
+    regularImage: ImageMetadata
+): SourceProps[] {
+    const props: SourceProps[] = []
+    if (smallImage) {
+        const smallSizes = getSizes(smallImage.originalWidth)
+        props.push({
+            media: "(max-width: 768px)",
```

Comment:
If I understand correctly, this fixes the main concern I had when looking at the code yesterday. To make sure I understand: The media attribute makes sure that the small image will only ever show up on small screen sizes (<= 768px), regardless of its own size? (This is important because I'll probably render mobile Grapher charts at a resolution that is close to the ideal one)

Reply:
Yes that's correct 🙂 

Which is why the mobile export can be 1536px wide for (DPR 2) but it will still only show on mobiles.